Sources/Workspace/InitPackage.swift

```diff
@@ -350,8 +378,58 @@ public final class InitPackage {
         }
     }
 
+    private func writePlugins() throws {
+        switch packageType {
+        case .buildToolPlugin, .commandPlugin:
+            let plugins = destinationPath.appending(component: "Plugins")
+            guard self.fileSystem.exists(plugins) == false else {
+                return
+            }
+            progressReporter?("Creating \(plugins.relative(to: destinationPath))/")
+            try makeDirectories(plugins)
+
+            let moduleDir = plugins.appending(component: "\(pkgname)")
+            try makeDirectories(moduleDir)
+
+            let sourceFileName = "plugin.swift"
+            let sourceFile = try AbsolutePath(validating: sourceFileName, relativeTo: moduleDir)
+
+            var content = """
+                import PackagePlugin
+
+                @main
+                """
+            if packageType == .buildToolPlugin {
+                content += """
+                    struct \(typeName): BuildToolPlugin {
+                        func createBuildCommands(context: PluginContext, target: Target) async throws -> [Command] {
+                            print("Hello, World!")
+                            return []
```

Comment:
A more full-fledged starting point would be to have a shared function between the SwiftPM and Xcode API, and some boilerplate to construct a sample build command.  Not sure how far to go in that direction.

Reply:
I think having a sample build command would be good. 

A bit less sure about how to handle Xcode vs. SwiftPM plugins, on the one hand, shared function is obviously the right thing for many use cases, on the other hand it could potentially make it confusing that the two entry points even exist?

Reply:
cc @bitjammer @TimTr for opinions on the content of these templates

Reply:
In the long run, it might make sense to have an option for whether or not to add the Xcode specifics.  We can leave it out for now but maybe include the URL of the plugins Markdown here in the SwiftPM repository, which talks about how to add them.  That would seem to be in line with how the argument parser template works (it puts in this comment: `// https://swiftpackageindex.com/apple/swift-argument-parser/documentation`).

Reply:
@tomerd good point that these seem to be using the older style of always having a `/Sources/<target>/` folder structure. With the updated templates for executables using just `/Sources/` the base templates can start a bit cleaner. Also note this PR: https://github.com/apple/swift-package-manager/pull/6294 which makes the SwiftPM folder search path work with `/Sources/` by default (no need for a `path:` entry), and also work with the traditional per-target folder model. So either option is now supported by default.

I don't have a strong opinion for these templates using the traditional or simplified model. But if the common case will be a single target, may be nice to stick with the simpler approach.

Reply:
Simplified model sounds good given that the manifest doesn't have to be made more complicated.  I'll update the PR.  Thanks!

Reply:
@TimTr What about the point about whether or not to include the optional XcodeProjectPlugin support to make it easier for new plugins to work with Xcode projects out-of-the-box.